vllm/model_executor/models/mllama.py

```diff
@@ -1016,6 +1027,11 @@ def forward(
         inputs_embeds = self.embed_tokens(input_ids)
         hidden_states = inputs_embeds
 
+        if is_hpu:
+            for idx, decoder_layer in enumerate(self.layers):
+                if isinstance(decoder_layer, LlamaDecoderLayer):
+                    self.layers[idx].self_attn.rotary_emb.prepare_cos_sin(
```

Comment:
Can we use logic introduced by https://github.com/HabanaAI/vllm-fork/pull/566 , and prepare_cos_sin inside HpuModelAdapterEncoderDecoder?

Reply:
Because the module architecture is not matching with that specified in `get_names_for_rope()`, so I met some failures. I have created a JIRA ticket to @tzielinski-habana , https://jira.habana-labs.com/browse/SW-212068